lib/util/Ordering.ts

```diff
@@ -0,0 +1,22 @@
+import * as RDF from 'rdf-js';
+import * as T from "../expressions/Term";
+import { transformLiteral } from "../Transformation";
+
+// Ordering 2 given strings
+export function orderTypes(litA: RDF.Term | undefined, litB: RDF.Term | undefined, isAscending:boolean){
+    if (litA && litA.termType === 'Literal' && litB && litB.termType === 'Literal') {
+        const a = transformLiteral(litA);
+        const b = transformLiteral(litB); 
+        return order(a, b, isAscending);
+    } else {
+        return 0;
+    }
+}
+
+// Effective ordering
+export function order(orderA: T.Literal<any>, orderB: T.Literal<any>, isAscending:boolean){
+    if (!orderA || !orderB || orderA.typedValue === orderB.typedValue) {
```

Comment:
As far as I can see, the output of `transformLiteral` is always defined. So I think these falsy checks can be removed, right?